common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationSupportSuite.java

```diff
@@ -709,12 +774,24 @@ public void testLocate() throws SparkException {
     assertLocate("大千", "test大千世界大千世界", 9, "UNICODE_CI", 9);
     assertLocate("大千", "大千世界大千世界", 1, "UNICODE_CI", 1);
     // Case-variable character length
+    assertLocate("\u0307", "i̇", 1, "UTF8_BINARY", 2);
```

Comment:
```suggestion
    assertLocate("\u0307", "i̇", 1, "UTF8_BINARY_LCASE", 2);
```

Reply:
I am not sure what the intent of this test case is here. If it is for UTF8_BINARY_LCASE then we have it again below.

Reply:
the intent is: show that "\u0307" is found in "i̇" on the binary level (with respect to UTF8_BINARY)

but "\u0307" is not found in "İ" on the character level (with respect to UTF8_BINARY_LCASE)